advanced/advanced-testing-examples/pom.xml

```diff
@@ -127,14 +171,14 @@
       <dependency>
         <groupId>org.springframework.integration</groupId>
         <artifactId>spring-integration-bom</artifactId>
-        <version>5.0.0.BUILD-SNAPSHOT</version>
+        <version>5.0.0.M6</version>
```

Comment:
Would you mind to jump back to the `5.0.0.BUILD-SNAPSHOT` for Spring Integration and remove exclude for the `jackson-module-kotlin`? We don't need it any more since the fix is done in the latest SI.
Thanks

Reply:
I think I prefer to keep the samples on a stable release.

Reply:
OK. Merging then...
I think the issue with `assemble` we should address in the separate commit:
```
FAILURE: Build failed with an exception.

* What went wrong:

Unable to find a single main class from the following candidates [org.springframework.integration.samples.barrier2.ErrorHandlingApplication, org.springframework.integration.samples.barrier.Application]
```